### ISSUE TITLE
Public interface broken in patch version (0.2.4)

### ISSUE THREAD

**Issue by futile** [ruabmbua/enet-sys#15]
The current version, 0.2.4, which contains the bindgen update from #14, is incompatible with 0.2.3.

Example:

`peerCount` in 0.2.3: https://docs.rs/enet-sys/0.2.3/enet_sys/struct._ENetHost.html#structfield.peerCount

```rust
peerCount: usize
```

`peerCount` in 0.2.4: https://docs.rs/enet-sys/0.2.4/enet_sys/struct._ENetHost.html#structfield.peerCount

```rust
peerCount: size_t // `size_t` resolves to `u64` (or `u32`)
```

Code that used `ENetHost::peerCount` for, e.g., the size/indexing of a vec will now fail, with something like "expected `usize`, found `u64`". Probably other fields/variables are affected as well, but this is one case that broke for me.

Because this was released as `0.2.4`, and not as `0.3`, this can break code that only runs `cargo update`.

Semver.org on this case ([link](https://semver.org/#what-do-i-do-if-i-accidentally-release-a-backwards-incompatible-change-as-a-minor-version)):

> **What do I do if I accidentally release a backwards incompatible change as a minor version?**
>
> As soon as you realize that you’ve broken the Semantic Versioning spec, fix the problem and release a new minor version that corrects the problem and restores backwards compatibility. Even under this circumstance, it is unacceptable to modify versioned releases. If it’s appropriate, document the offending version and inform your users of the problem so that they are aware of the offending version.

(Since the major version is still 0, "minor version" here refers to the "patch version" I believe).

So maybe publishing the current state as 0.3, and either yanking 0.2.4  or publishing 0.2.5 with the changes reverted would be a solution? Or maybe adjusting bindgen to use other/the same as before types might also be an option. But not sure, just my initial ideas.

Thanks for the quick reply to #14 btw.!

**Comment by ruabmbua** [comment]
Hm, since I did not change the c lib version I guess this is some cross platform incompatibility. I generated this struct on x86_64 linux and just copied it in. I guess the struct uses some platform dependent fields / field types.

So this is most likely just a bug, will look into it.

**Comment by ruabmbua** [comment]
In the meantime I am going to yank the newly released version.

**Comment by futile** [comment]
> Hm, since I did not change the c lib version I guess this is some cross platform incompatibility. I generated this struct on x86_64 linux and just copied it in. I guess the struct uses some platform dependent fields / field types.
> 
> So this is most likely just a bug, will look into it.

I used this on x86_64 linux, so that should at least be the same platform. I would think that bindgen simply changed how it generates bindings for C's `unsigned long`: It used `usize` before, and now uses `u64` (or `u32`) explicitly.

**Comment by ruabmbua** [comment]
Ah you are right, was a bit suspicious about the manually added struct.

**Comment by ruabmbua** [comment]
Published a v1.0.0 to crates.io.
